frontend/src/features/public-form/components/FormFields/PublicFormSubmitButton.tsx

```diff
@@ -79,9 +86,15 @@ export const PublicFormSubmitButton = ({
     }
   }
 
+  const isSingleSubmissionOnlyModalOpen = hasSingleSubmissionValidationError
+  const onSingleSubmissionModalClose = () => {
+    setHasSingleSubmissionValidationError(false)
+  }
+
   return (
     <Stack px={{ base: '1rem', md: 0 }} pt="2.5rem" pb="4rem">
-      {isOpen ? (
+      // Prevent opening of multiple modals
+      {isOpen && !isSingleSubmissionOnlyModalOpen ? (
```

Comment:
This shouldn't be possible iirc. The payment modals should close before triggering the submission API.

Reply:
regarding this, i was thinking that it might be good to still have the `isPaymentsModalOpen && !isSingleSubmissionOnlyModalOpen` check, since it would be more defensive. 

any thoughts on this?

Reply:
I think we should not need to put defensive code here, which might be swallowing mistakes rather than surfacing them up. i.e., someone made a mistake of not dismissing their modal, but had no idea that this is happening due to this piece of code "helping to hide" the mistake.

Reply:
fair enough, i will remove this check